Automattic/wp-calypso
# Blog onboarding: Update celebration screen to support design-first flow

Related to 2522-gh-Automattic/dotcom-forge

## Proposed Changes

- Make sure we don't have the gray line at the top
- Change the subtitle according to the next action (Connect to social or writing the first post)
- Preview has enough size to show the site
- Ensure selected domain appears (if user selected one)
- Define what to do on preview in case user didn't write a first post. Started a [thread about it here](p1684879946959299/1684879942.447799-slack-CKZHG0QCR).
- If user published the first post show Connect to social button, otherwise show Write your first post CTA
- Create a test for it

## Testing Instructions



- Load [this Jetpack PR](https://github.com/Automattic/jetpack/pull/30871) to your WPCOM Sandbox, but since the Jetpack PR is now merged, if you can't load it with the command above, you can use this diff instead D111566-code
- Go to /setup/design-first on an account with 0 existing sites.
- Launch a free site with no custom domain
- Run unit tests with `yarn test-client client/landing/stepper/declarative-flow/internals/steps-repository/celebration-step/test/index.test.tsx`
- Ensure `start-writing` celebration screen also works as expected, since we're reusing that code.
- Ensure we have `Now it’s time to start posting.` as subtitle and `Write your first post` CTA when user didn't write a post.
- Create and publish a post for the current site in a new tab, refresh the celebration screen and you should see `Now it’s time to connect your social accounts.` as subtitle and `Connect to social`. 

 

## Pre-merge Checklist
